src/Faker/Provider/Base.php

```diff
@@ -62,6 +62,24 @@ public static function randomDigitNot($except)
         return $result;
     }
 
+    /**
+     * Generates a random digit, which cannot be in $except
+     *
+     * @param array|int $except
+     * @return int
+     */
+    public static function randomDigitNotIn($except)
+    {
+        $digits = array(0, 1, 2, 3, 4, 5, 6, 7, 8, 9);
+        $pool = array_diff($digits, (array) $except);
+        if (!$pool) {
+            throw new \InvalidArgumentException('The argument is inclusive of the digits 0 to 9. A digit cannot be returned.');
+        }
+        $key = array_rand($pool);
```

Comment:
prior to php 7.1, this does not use a Mersenne Twister randomizer. Use the `randomElement` formatter to choose an element in the array.

Reply:
Change made as requested. The test failure is unrelated:
```
Faker\Test\Provider\ImageTest::testDownloadWithDefaults
getimagesize(): Read error!
```